VideoSyncClass/VideoSync.py

```diff
@@ -0,0 +1,283 @@
+import numpy as np
+import matplotlib.pyplot as plt
+from subprocess import run, call
+from pathlib import Path
+import yaml
+import os
+from scipy.io import wavfile
+from scipy.io.wavfile import read as wavread
+import mne
+
+class VideoSync:
+    def __init__(self, input_folder=None, output_folder="/output"):
+        """
+        Summary: VideoSync represents an object performing synchronization between multiple streams of data.
+
+        Attributes:
+            input_folder: Directory where data is supplied from.
+            output_folder: Directory where analysis files will be published to. Defaults to "input_folder/output"
+
+            self.video_paths: List of RELATIVE paths to video files.
+            self.audio_paths: List of RELATIVE paths to audio files.
+            self.meg_paths: List of RELATIVE paths to meg files.
+        """
+        # User defined paths for input/output directories
+        self.__input_folder = input_folder
+        self.__output_folder = output_folder
+
+        # Paths of data within a folder
+        self.__video_paths = []
+        self.__audio_paths = []
+        self.__meg_paths= []
+
+        # Instance variables representing data
+
+
+    def get_video_paths(self):
+        """
+        Returns:
+            None: if no path has been defined
+            string: List of paths (strings) representing videos
+        """
+        if len(self.__video_paths) == 0:
+            print("video_path has not yet been instantiated. Please use set_video_path or parse_files_from_folder to instantiate this attribute.")
+            return None
```

Comment:
several cases like this below; I would have expected this to raise an error rather than return `None`, but maybe I'm not understanding / envisioning the usage clearly/correctly.